stream.go

```diff
@@ -172,23 +186,38 @@ func (cs *clientStream) RecvMsg(m interface{}) (err error) {
 			return toRPCErr(errors.New("grpc: client streaming protocol violation: get <nil>, want <EOF>"))
 		}
 		if err == io.EOF {
+			if EnableTracing {
+				defer cs.traceInfo.tr.Finish()
+			}
 			if cs.s.StatusCode() == codes.OK {
 				return nil
 			}
 			return Errorf(cs.s.StatusCode(), cs.s.StatusDesc())
 		}
+		if EnableTracing {
+			cs.traceInfo.tr.LazyLog(&fmtStringer{"%v", []interface{}{err}}, true)
+			cs.traceInfo.tr.SetError()
+		}
 		return toRPCErr(err)
 	}
 	if _, ok := err.(transport.ConnectionError); !ok {
 		cs.t.CloseStream(cs.s, err)
 	}
 	if err == io.EOF {
+		if EnableTracing {
+			cs.traceInfo.tr.Finish()
```

Comment:
yeah, this is a sure sign that the handling of Finish is wrong in here. There should be one guaranteed path to get to the completion of the trace, but you've scattered this all over the place.